app/models/case_court_report.rb

```diff
@@ -88,7 +88,15 @@ def aggregate_contact_dates(people)
       contact_dates[contact_type] = [date_with_format]
     end
 
-    contact_dates
+    sort_dates(contact_dates)
+  end
+
+  def sort_dates(contact_dates)
+    contact_dates.each_value do |dates|
+      next if dates.size <= 1
+
+      dates.sort! { |first_date, second_date| first_date.delete("*") <=> second_date.delete("*") }
+    end
```

Comment:
Can I help you write a spec for this? Did you have something that you tried but didn't work that you can push up?

Reply:
Hi @littleforest thanks a lot. I tried to test/find in the spec withing the report, that the dates  appear sorted. But I got lost while searching. 😞

Reply:
@josearmandojacq I thought it wouldn't be too difficult to test this method, but I just noticed that this a private method, so it shouldn't be tested directly. The `case_court_report` spec is a little bit hard to follow - can you add an expectation after [this line](https://github.com/rubyforgood/casa/blob/main/spec/models/case_court_report_spec.rb#L79)? (Are the dates on line 73-75 what you are trying to sort, or am I looking at the wrong place?) I am not very familiar with this piece of the code, so I might not be able to be as helpful as I thought I could.

Reply:
If there is a private method that we want to test directly, we could break it out to a service class and test that new class directly

Reply:
@littleforest, @compwron  It is not about testing a private method, but instead testing that in the report the dates appear ordered ascendant. I tried to do it while looking into the string/xml created in the test, but it was a lot and couldn't find the proper way to test it. 😁

Reply:
Happy you are commenting on it 💪🏽 💪🏽

Reply:
I will try again in these days because I don't have too much time now. But feel free @compwron to merge it if you think it is good enough based in the priority :)